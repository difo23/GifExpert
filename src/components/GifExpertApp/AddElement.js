import React, { useState } from 'react';
import PropTypes from 'prop-types';
const AddElement = ({ handleAdd }) => {
	const [ element, setElement ] = useState('');

	const handleChange = (e) => {
		setElement(e.target.value);
	};

	const handleSumit = (e) => {
		e.preventDefault();

		if (element.trim().length > 2) {
			handleAdd((category) => [ ...category, element ]);
		}

		setElement('');
	};

	return (
		<div>
			<p>{element}</p>
			<form onSubmit={handleSumit}>
				<div>
					<input type="text" value={element} onChange={handleChange} />
				</div>
			</form>
		</div>
	);
};

AddElement.propTypes = {
	handleAdd: PropTypes.func.isRequired
};

export default AddElement;
