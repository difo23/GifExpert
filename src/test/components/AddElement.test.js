import React from 'react';
import { shallow } from 'enzyme';
import AddElement from '../../components/GifExpertApp/AddElement';
import { Simulate } from 'react-dom/test-utils';

describe('Pruebas del componente <AddElement />', () => {
	const handleAdd = jest.fn();

	let wrapper = shallow(<AddElement handleAdd={handleAdd} />);

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddElement handleAdd={handleAdd} />);
	});

	test('should snapshot verify', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should simulate change', () => {
		const value = 'text value';

		wrapper.find('input').simulate('change', { target: { value } });

		const p = wrapper.find('p');

		expect(p.text().trim()).toBe(value);
	});

	test('should simulate submit no llamar la funcion handleAdd', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(handleAdd).not.toHaveBeenCalled();
	});

	test('should simulate submit llamando la funcion handleAdd y limpiando el input element', () => {
		const value = 'text value';

		wrapper.find('input').simulate('change', { target: { value } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(handleAdd).toHaveBeenCalled();
		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
