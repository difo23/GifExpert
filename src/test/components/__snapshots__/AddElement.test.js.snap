// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <AddElement /> should snapshot verify 1`] = `
<div>
  <p />
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
